src/lib/config/ModelConfig.cc

```diff
@@ -0,0 +1,997 @@
+        configTree_["Core"]["ISA"]);
+    std::string ISA;
+    configTree_["Core"]["ISA"] >> ISA;
+    if (result.errored) {
+      std::cerr << "[SimEng:ModelConfig] Invalid ISA value of \"" << ISA
+                << "\" passed in config file due to \"" << result.message
+                << "\" error. Cannot continue with config validation. Exiting."
+                << std::endl;
+      exit(1);
+    }
+    // Set ISA_
+    if (ISA == "AArch64") {
+      ISA_ = ISA::AArch64;
+    } else if ("rv64") {
+      ISA_ = ISA::RV64;
+    }
+  }
+  createGroupMapping();
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<std::string>("emulation",
+                                                      "Simulation-Mode"));
+  expectations_["Core"]["Simulation-Mode"].setValueSet(
+      std::vector<std::string>{"emulation", "inorderpipelined", "outoforder"});
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<float>(1.f, "Clock-Frequency"));
+  expectations_["Core"]["Clock-Frequency"].setValueBounds(0.f, 10.f);
+
+  // Early check on ["Core"]["Clock-Frequency"] as values are needed to inform
+  // the expected lower bound of the ["Core"]["Timer-Frequency"] value
+  uint64_t tFreqUpperBound = 1000;
+  if (!isDefault) {
+    ValidationResult result =
+        expectations_["Core"]["Clock-Frequency"].validateConfigNode(
+            configTree_["Core"]["Clock-Frequency"]);
+    float cFreq;
+    configTree_["Core"]["Clock-Frequency"] >> cFreq;
+    if (result.errored) {
+      std::cerr << "[SimEng:ModelConfig] Invalid Clock-Frequency value of \""
+                << cFreq << "\" passed in config file due to \""
+                << result.message
+                << "\" error. Cannot continue with config validation. Exiting."
+                << std::endl;
+      exit(1);
+    }
+
+    tFreqUpperBound = cFreq * 1000;
+  }
+
+  expectations_["Core"].addChild(
+      ExpectationNode::createExpectation<uint64_t>(100, "Timer-Frequency"));
+  expectations_["Core"]["Timer-Frequency"].setValueBounds<uint64_t>(
+      1, tFreqUpperBound);
+
+  expectations_["Core"].addChild(ExpectationNode::createExpectation<bool>(
+      false, "Micro-Operations", true));
+  expectations_["Core"]["Micro-Operations"].setValueSet(
+      std::vector{false, true});
+
+  if (ISA_ == ISA::AArch64) {
+    expectations_["Core"].addChild(ExpectationNode::createExpectation<uint64_t>(
+        512, "Vector-Length", true));
+    expectations_["Core"]["Vector-Length"].setValueSet(
+        std::vector<uint64_t>{128, 256, 384, 512, 640, 768, 896, 1024, 1152,
+                              1280, 1408, 1536, 1664, 1792, 1920, 2048});
+
+    expectations_["Core"].addChild(ExpectationNode::createExpectation<uint64_t>(
+        512, "Streaming-Vector-Length", true));
+    expectations_["Core"]["Streaming-Vector-Length"].setValueSet(
+        std::vector<uint64_t>{128, 256, 384, 512, 1024, 2048});
+  }
```

Comment:
I feel as though the default values of VL and SVL should be 128, as it makes more sense to go with the lower bound rather than a64fx's value of VL. (Won't die on this hill...)